pkg/cloud/services/compute/instance.go

```diff
@@ -470,12 +471,19 @@ func (s *Service) getOrCreatePort(eventObject runtime.Object, clusterName string
 		SecurityGroups:      securityGroups,
 		AllowedAddressPairs: []ports.AddressPair{},
 	}
-
-	for _, ap := range portOpts.AllowedAddressPairs {
-		createOpts.AllowedAddressPairs = append(createOpts.AllowedAddressPairs, ports.AddressPair{
-			IPAddress:  ap.IPAddress,
-			MACAddress: ap.MACAddress,
-		})
+	// do not add allowed-address-pairs
+	if portOpts.DisablePortLevelSecurity == false {
+		for _, ap := range portOpts.AllowedAddressPairs {
+			createOpts.AllowedAddressPairs = append(createOpts.AllowedAddressPairs, ports.AddressPair{
+				IPAddress:  ap.IPAddress,
+				MACAddress: ap.MACAddress,
+			})
+		}
+	}
+	// do not add security groups
```

Comment:
I think the comment should go a line below and for clarity should be something like: "when port security is disabled, add neither instance nor port level security groups to the port".

Reply:
ok, will do that.